eth/backend_test.go

```diff
@@ -45,3 +50,46 @@ func TestNewTxLog(t *testing.T) {
 	txLog, err = b.newTxLog(tx)
 	assert.EqualError(err, "no method signature")
 }
+
+func TestSendTransaction_SendErr_DontUpdateNonce(t *testing.T) {
+	// client := &ethclient.Client{}
+	client, err := ethclient.Dial("https://rinkeby.infura.io")
```

Comment:
Looks like there might be a way to generate a test chain - see the [ethclient tests](https://github.com/ethereum/go-ethereum/blob/2de37f28e069b059d6a8fcf546ecf307373e2419/ethclient/ethclient_test.go#L187) from go-ethereum. Haven't tried it, but if it works this could make writing/running this test easier since the ethclient instance wouldn't need to interact with an actual ETH RPC provider.

Reply:
Sadly they are unexported and they seem to be more "chain/block propagation" related whereas we need a simple RPC error. 

Given the urgency for this fix would you be okay keeping the test as is and exploring a more robust solution for this in the future? 

I can see this end up being flaky in the following cases:
- There is no internet connection 
- The endpoint is down ( == the entire rinkeby cluster for infura is down)

Reply:
Ah actually I just realized that a requirement for this test is to trigger an error when `ethclient.SendTransaction()` is called and this is already accomplished because Infura will return an error due to a missing project ID. Sounds good to me!